File: Semester 4/Advanced Programming/RpcProtocolC#/Repository/ICrudRepository.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Domain;
namespace Repositories
{
    public interface ICrudRepository<ID, E> where E : Entity<ID>
    {
        void Save(E entity);
        void Delete(ID id);
        void Update(E entity);
        E FindOne(ID id);
        IEnumerable<E> FindAll();
    }
}

File: Semester 4/Advanced Programming/RpcProtocolJava/ApplicationRepository/src/main/java/app/repository/ICrudRepository.java
package app.repository;

import app.model.Entity;

import java.util.List;

public interface ICrudRepository<ID, E extends Entity<ID>> {
    void save(E entity);
    void delete(ID id);
    void update(E entity);
    E findOne(ID id);
    List<E> findAll();
}